Sentry Performance Tracing

mpass99 opened the pull request:
Relates to #270 

With [Sentry's Distributed Tracing](https://docs.sentry.io/product/sentry-basics/tracing/distributed-tracing/) we can connect the Tracing of our applications to get a more detailed view into the time behavior.

For Poseidon we just have to enable the tracing functionality of Sentry. For that see also: [38ffc3e9](https://lab.xikolo.de/codeocean/codeocean-terraform/-/commit/38ffc3e9247abddf55191bf951569f6e79cd433d).

The tracing connection between the application is done automatically by Sentry (via the `trace-id` http header; [See](https://docs.sentry.io/platforms/ruby/performance/connect-services/)).

With only this we would still have two separate transactions - one for Submission create and one for Submission Score. To join them we can enable the Sentry Tracing for the JavaScript Frontend.

MrSerth commented:
Awesome, thanks for getting that started! I was reviewing commit-by-commit and find all changes plausible. Besides reviewing, I also started having a look at Sentry and was happy to see some traces there ;).

What I was still thinking are two things regarding the spans:

1. Should we add additional spans, e.g., in some filters running with every request or other often-used methods?
   This would allow us to see more details, such as shown in [this trace](https://sentry.io/organizations/openhpi/performance/codeocean:57c4621bae8c410090330525a21596c1).
2. Can we add the current (custom) span names into the "description" field and rather an "identifier" as the current name? This would allow to see the name for spans. For example, compare [this custom span](https://sentry.io/organizations/openhpi/performance/summary/spans/List%20File%20System:d41d8cd98f00b204/?project=6077585&transaction=GET+%2Fapi%2Fv1%2Frunners%2F%2A%2Ffiles) to [this pre-defined span](https://sentry.io/organizations/openhpi/performance/summary/spans/http.server:ea72d7f241df4722/?project=6077585&query=http.method%3APATCH&statsPeriod=24h&transaction=PATCH+%2Fapi%2Fv1%2Frunners%2F%2A%2Ffiles). We might also come up with an own naming scheme, e.g., `poseidon.operation`, `poseidon.manager`, `poseidon.storage`, `poseidon.nomad.execute`, `influx.send`, `nomad.query`, `response.render`, ...

mpass99 commented:
Good remarks!  
I added an identifier for the custom spans.  
Also, I added more custom spans for the most used and most time consuming functionality, `Execute`. If you would like to know the time behavior of other methods, please let me know.

MrSerth reviewed (approved):
Awesome! I was just testing the new tracing with CodeOcean, and it works like a charm!

Let's evaluate the responses once we have some data to identify where to add more details. :)
